run_linux.go

```diff
@@ -269,8 +269,7 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 		bindFiles[config.DefaultHostsFile] = hostFile
 	}
 
-	// generate /etc/hostname if the user intentionally did not override
-	if !(contains(volumes, "/etc/hostname")) {
+	if !options.NoHostname && !(contains(volumes, "/etc/hostname")) {
 		if _, ok := bindFiles["/etc/hostname"]; !ok {
```

Comment:
I don't see how `ok` can ever be true here.  Did I miss something?

Reply:
I think this is for the use case where a user does `-v /tmp/hostname:/etc/hostname`

Reply:
I still don't follow.  If I'm reading it right, the `bindFiles` map has, at most, one item in it: a value for the `config.DefaultHostsFile` key.  A `-v` flag from the command line would show up in `options.Volumes` or `b.CommonBuildOpts.Volumes`, or maybe both if we're in `build`.

`contains()` probably shouldn't exist when we got rid of for `github.com/containers/buildah/util.StringInSlice()` in favor of `github.com/containers/common/pkg/util.StringInSlice()`.

Reply:
That was existing code, so I will remove the check.